taichi/codegen/spirv/kernel_utils.cpp

```diff
@@ -99,8 +90,8 @@ KernelContextAttributes::KernelContextAttributes(
     for (int i = 0; i < vec->size(); ++i) {
       auto &attribs = (*vec)[i];
       const size_t dt_bytes =
-          (attribs.is_array && !is_ret && has_buffer_ptr)
-              ? sizeof(uint64_t)
+          (attribs.is_array && !is_ret)
+              ? (has_buffer_ptr ? sizeof(uint64_t) : sizeof(uint32_t))
```

Comment:
What's `sizeof(uint32_t)` for here?

Reply:
When `has_buffer_ptr` is false, we need a placeholder argument. It was the element type of the pointer before. In this PR I changed it to u32 for simplicity.

Reply:
BTW we no longer use this function to calculate the offset of the arguments anymore. This function still calculates the offset of rets. We will remove them along with `ArgAttribute` and `RetAttribute` as soon as the refactor completes.